### PR TITLE
fix[test] integration tests

### LINKED ISSUE #202: Catch HTTPError where applicable
When using Alchemy RPC, this line is raising HTTPError (400 Client Error: Bad Request)
https://github.com/vyperlang/titanoboa/blob/786e44fa1bb061f19118f9985c6343bad6e0f2b2/boa/rpc.py#L121

But we only catch RPCError here
https://github.com/vyperlang/titanoboa/blob/192cb293f026bd2f06e60a05c541005152c35aeb/boa/network.py#L127

let's do a quick search for everywhere that catches RPCError though and see if we also should catch HTTPError

### PR DESCRIPTION
**Pull request by DanielSchiavini**
Fixes #202
Tested in https://github.com/DanielSchiavini/titanoboa/actions/runs/9029296510/job/24811369660?pr=7 via https://github.com/DanielSchiavini/titanoboa/pull/7

### Cute Animal Picture

image

### CODE REVIEW

File: boa/contracts/base_evm_contract.py [hunk 1]
```diff
@@ -45,6 +45,12 @@ class StackTrace(list):
     def __str__(self):
         return "\n\n".join(str(x) for x in self)
 
+    @property
+    def dev_reason(self) -> str | None:
+        if isinstance(self.last_frame, str):
```
Comment (charles-cooper) [inline boa/contracts/base_evm_contract.py:50]: oof, when can this happen?

File: boa/contracts/base_evm_contract.py [hunk 2]
```diff
@@ -76,9 +82,9 @@ def _handle_child_trace(computation, env, return_trace):
     else:
         child_trace = child_obj.stack_trace(child)
 
-    if child_trace.last_frame.dev_reason and not return_trace.last_frame.dev_reason:
+    if child_trace.dev_reason and not return_trace.dev_reason:
```
Comment (charles-cooper) [inline boa/contracts/base_evm_contract.py:85]: this change looks spooky. maybe we should check if `last_frame is not None`?

File: boa/network.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def _get_capability(self, hex_bytecode):
         try:
             self._rpc.fetch("eth_call", [{"to": None, "data": hex_bytecode}])
             return True
-        except RPCError:
+        except (RPCError, HTTPError):
```
Comment (charles-cooper) [inline boa/network.py:95]: `eth_call` can raise an HTTPError now?

File: boa/test/plugin.py [hunk 4]
```diff
@@ -56,6 +56,9 @@ def pytest_collection_modifyitems(config, items):
 
 
 def pytest_fixture_setup(fixturedef, request):
+    if request.node.get_closest_marker("ignore_isolation"):
```
Comment (charles-cooper) [inline boa/test/plugin.py:59]: i'm just going to disable this for now actually

Comment (charles-cooper) [reply boa/test/plugin.py:59]: never mind, see https://github.com/vyperlang/titanoboa/pull/219

File: boa/contracts/base_evm_contract.py (hunk 1 above)
Comment (DanielSchiavini) [reply boa/contracts/base_evm_contract.py:50]: a couple lines later: https://github.com/DanielSchiavini/titanoboa/blob/18ab5b64dfe6ef6e41fc708fea9263c52e54a461/boa/contracts/base_evm_contract.py#L61

File: boa/contracts/base_evm_contract.py (hunk 2 above)
Comment (DanielSchiavini) [reply boa/contracts/base_evm_contract.py:85]: ditto

File: boa/network.py (hunk 3 above)
Comment (DanielSchiavini) [reply boa/network.py:95]: :thinking: maybe not

Comment (DanielSchiavini) [reply boa/network.py:95]: Yeah it shouldn't, this is not a `debug_` call

Comment (DanielSchiavini) [reply boa/network.py:95]:
```suggestion
        except RPCError:
```

File: boa/contracts/base_evm_contract.py (hunk 2 above)
Comment (charles-cooper) [reply boa/contracts/base_evm_contract.py:85]: we should still test for `is not None`, as you could have an empty dev reason (which is valid)

Comment (DanielSchiavini) [reply boa/contracts/base_evm_contract.py:87]: https://github.com/vyperlang/titanoboa/pull/213/commits/b5b1f9e3fa97f80a3278157a8b721019b9a30c8f

Comment (charles-cooper) [reply boa/contracts/base_evm_contract.py:87]: the return value of `dev_reason` could still be an empty string!

Comment (DanielSchiavini) [reply boa/contracts/base_evm_contract.py]: https://github.com/vyperlang/titanoboa/pull/213/commits/0ca0ea7459f54ac7d69d57e944a1fc9a65422158

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**